pkg/provider/aci_test.go

```diff
@@ -1582,3 +1582,168 @@ func TestGetImagePullSecretsWithDockerCfgSecret(t *testing.T) {
 		})
 	}
 }
+
+func TestGetImagePullSecretsWithDockerConfigJSONSecret(t *testing.T) {
+	podName := "pod-" + uuid.New().String()
+	podNamespace := "ns-" + uuid.New().String()
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	aciMocks := createNewACIMock()
+
+	pod := testsutil.CreatePodObj(podName, podNamespace)
+
+	invalidSecretType := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fakeSecret",
+			Namespace: podNamespace,
+		},
+		Type: "fakeType",
+	}
+
+	invalidSecretNoDockerConfigJsonKey := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fakeSecret",
+			Namespace: podNamespace,
+		},
+		Type: v1.SecretTypeDockerConfigJson,
+	}
+
+	invalidCfgJson := `{
+		"repoData": {
+			"auths": {}
+		}
+	}`
+	invalidSecretMalformedCfgJson := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fakeSecret",
+			Namespace: podNamespace,
+		},
+		Type: v1.SecretTypeDockerConfigJson,
+		Data: map[string][]byte{
+			v1.DockerConfigJsonKey: []byte(invalidCfgJson),
+		},
+	}
+
+	validCfgJson := `{
+		"auths": {
+			"repoData": {
+				"username": "fakeUserName",
+				"password": "fakePassword"
+			}
+		}
+	}`
+	validSecretCfgJson := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fakeSecret",
+			Namespace: podNamespace,
+		},
+		Type: v1.SecretTypeDockerConfigJson,
+		Data: map[string][]byte{
+			v1.DockerConfigJsonKey: []byte(validCfgJson),
+		},
+	}
+
+	cases := []struct {
+		description      string
+		imagePullSecrets []v1.LocalObjectReference
```

Comment:
I see imagePullSecrets is the same for all test cases. You can remove this variable and make it a const used in line 1727.